src/common/snippets/src/lowered/pass/split_loops.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/split_loops.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+using LoopManager = LinearIR::LoopManager;
+using LoopInfoPtr = LoopManager::LoopInfoPtr;
+
+SplitLoops::SplitLoops() : Pass() {}
+
+bool SplitLoops::can_be_split(const LoopInfoPtr& current, const LoopInfoPtr& parent) {
+    return current->work_amount == parent->work_amount && current->dim_idx == parent->dim_idx &&
+           current->increment != parent->increment;
+}
+
+bool SplitLoops::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SplitLoops")
+    if (linear_ir.empty())
+        return false;
+
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    bool loop_was_split = false;
+    for (const auto& expr : linear_ir) {
+        const auto& loop_ids = expr->get_loop_ids();
+        if (loop_ids.empty())
+            continue;
+
+        // Note: we currently consider only the outermost loops for splitting
```

Comment:
Seems like we can make the pass more general. May I ask you to create the ticket for that as well? 🙂

Reply:
The ticket has been created: 113755. Thanks!